ui/src/components/common/Holdings.tsx

```diff
@@ -37,9 +37,10 @@ const Holdings: React.FC<Props> = ({ deposits, exchanges, role, sideNav, onLogou
                     { deposits.map(deposit => {
                         const { asset, account } = deposit.contractData;
                         const options = exchanges.map(exchange => {
+                            const exchangeParty = exchange.contractData.exchange;
                             return {
                                 key: exchange.contractId,
-                                text: exchange.contractData.exchange,
+                                text: exchange.registryData?.name + ' (' + exchangeParty + ')' || exchangeParty,
```

Comment:
This does work, since `||` doesn't necessarily need to return a bool - if the left hand value is false or undefined, it will return whatever the right hand value is.

It's a bit weird due to JavaScript nonsense, since if the left hand is a 'falsy' value (such as an empty string) it will consider that false and return the right hand value. In this case, that may be the preferred behavior. If it's an empty string, we might want to show the party name. Let me know what you think about that.

Either way, I can switch to the pattern you suggested if you prefer, and I'll definitely switch to use the ` string construction syntax.

Reply:
Forget what I was saying about `||`. I realize now the problem is with adding a string to the `exchange.registryData?.name`. It would still be good to figure out if we want to show the party name if the string is empty though.

Reply:
hm my confusion was that an undefined `registryData` will concatenate an undefined to the `(exchangeParty)` string resulting in the `(exchangeParty)` string  which then will never be `undefined` (so as to trigger the right hand side of `exchangeParty`. 
If `undefined + string = undefined` then you are correct. Just don't know the exact behavior of this js nuance.

Reply:
I just tested it, it is indeed `undefined + "string" = "undefinedstring" because JavaScript is ridiculous.

Reply:
My favorite is 

```javascript
('b' + 'a' + + 'a' + 'a').toLowerCase()
-> "banana"
```